build-release.sh

```diff
@@ -0,0 +1,44 @@
+set -e
+
+(
+    cd modality-ros-hook
+    cargo clippy --all-features -- -W clippy::all -D warnings
+    cargo fmt --all -- --check
+    cargo test --all-features -- --test-threads=1
+    cargo build --profile=release-with-debug
+)
+
+mkdir -p target/release
+
+gcc -Wall -fPIC -shared \
```

Comment:
We should add `-Werror` to treat all warnings as errors at least.
Some possibly useful extra warnings (that may not be included via `-Wall` or `-Wextra`):
* `-Wextra`
* `-Wbad-function-cast`
* `-Wstrict-prototypes`
* `-Wmissing-declarations`
* `-Wmissing-include-dirs`
* `-Wmissing-prototypes`
* `-Wshadow`